### PR TITLE
fix: render indentation guides for Python challenges

### LINKED ISSUE #52406: Indentation Markers/Assistance in Code Editor for Python
### Is your feature request related to a problem? Please describe.

When writing Python code in the editor, it would can be a bit frustrating for campers to keep track of the indentation specially if the instructions are a bit wordy. Also, the tests would fail in case indentation is incorrect. 

### Describe the solution you'd like

It would be a good feature to provide assistance within the browser editor to maintain indentation.

### Describe alternatives you've considered

A good example is how VSCode provides markers:

image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52406

### CODE REVIEW

**Review by huyenltnguyen — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 

P.S. Sem, you really need to update your fork's `main` branch. I got a few old errors, because your branch is off of October.
